### PR TITLE
Add scope information to generated types

### PR DESCRIPTION
**Pull request by anthony-c-martin**

### CODE REVIEW

**Comment by majastrz** [comment]
> * **Valid Scope(s)**: Unknown

I think we're going to have to revisit proposing that swagger extension to make this more accurate in the future.

---
Refers to: generated/blueprint/microsoft.blueprint/2018-11-01-preview/types.md:4 in d4102c1. [](commit_id = d4102c14296ef3f1f060914cb26b7c5fe6fa0849, deletion_comment = False)

**Comment by majastrz** [comment]
> * **Additional Properties Type**: string

CodeFlow choked on this PR. This is the last one I see, but GitHub shows more :(

---
Refers to: generated/EnterpriseKnowledgeGraph/microsoft.enterpriseknowledgegraph/2018-12-03/types.md:27 in d4102c1. [](commit_id = d4102c14296ef3f1f060914cb26b7c5fe6fa0849, deletion_comment = False)

File: src/Bicep.Types/Concrete/ScopeType.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+        Tenant = 1 << 0,
+
+        ManagementGroup = 1 << 1,
+
+        Subscription = 1 << 2,
+
+        ResourceGroup = 1 << 3,
+
+        Extension = 1 << 4,
```
Comment (majastrz) [inline src/Bicep.Types/Concrete/ScopeType.cs:20]: Probably not super important, but do we need to model extensions of extensions as well?

**Review by majastrz — APPROVED** [review]

**Comment by anthony-c-martin** [comment]
> > * **Valid Scope(s)**: Unknown
> 
> I think we're going to have to revisit proposing that swagger extension to make this more accurate in the future.

Agreed. For now, I've been collecting information for ambiguous scoped types here, which we could also consider using in this project: https://github.com/Azure/azure-resource-manager-schemas/blob/master/generator/autogenlist.ts#L80-L97

File: src/Bicep.Types/Concrete/ScopeType.cs (hunk 1 above)
Comment (anthony-c-martin) [reply src/Bicep.Types/Concrete/ScopeType.cs:20]: Extensions of extensions should just show up as 'ScopeType.Extension' - I'm not differentiating.

**Pull request merged**